scrapy_poet/page_input_providers.py

```diff
@@ -133,6 +122,17 @@ def is_provided(self, type_: Callable) -> bool:
                 f"{self!r}. Expected either 'set' or 'callable'"
             )
 
+    @property
+    def has_cache_support(self):
```

Comment:
This, and the code that uses it, assumes that if one dep is serializable then all of them are. Maybe we even don't need this function and should just check the deps directly when deciding whether to tryu caching them.